sql/table.cc

```diff
@@ -8716,7 +8716,15 @@ bool TR_table::query_sees(bool &result, ulonglong trx_id1, ulonglong trx_id0,
 
   if (trx_id1 == ULONGLONG_MAX || trx_id0 == 0)
   {
-    result= true;
+    if (trx_id0 != trx_id1)
```

Comment:
eh, isn't that guaranteed? if trx_id0 == trx_id1 you return just few lines above.

Reply:
No, it's not guaranteed. It's user-defined.

Reply:
Sure. Fixed.